core-bundle/src/EventListener/DataContainer/PreviewLinkListener.php

```diff
@@ -112,7 +112,8 @@ public function createFromUrl(DataContainer $dc): void
                 case 'toggle':
                 case 'delete':
                 default:
-                    if ($dc->activeRecord->createdBy !== $userId) {
+                    $createdBy = $this->connection->fetchOne('SELECT createdBy FROM tl_preview_link WHERE id=?', [$dc->id]);
```

Comment:
Without casting, the comparison below will always be true. I'm not sure what should happen if `$userId` was set to `0` in L84.

```suggestion
                    $createdBy = (int) $this->connection->fetchOne('SELECT createdBy FROM tl_preview_link WHERE id=?', [$dc->id]);
                    
```

Reply:
I don't think that's true, because `$user->id` is a string as well. Obviously a type-safe comparison would be better, but we'd need to cast the values everywhere.

Reply:
If `$user->id` is always a string, two things should probably be adjusted:

*  the `@property` annotation in `BackendUser` (or more specific `User`) says `string|integer`
* L84 of this class assigns an integer as fallback value

Reply:
`$user->id` can be an integer - for instance if you manually created the object instance with a custom model.

In any case I would cast to `int` here for defensive programming.

Reply:
See See 8116179273ebda6ffca7ab6b316992d6e98624ff